lib/bulk/common.js

```diff
@@ -951,6 +952,9 @@ class BulkOperationBase {
       };
 
       if (op[key].hint) {
+        if (maxWireVersion(this.s.topology) < 5) {
+          throw new MongoError(`The current topology does not support a hint on ${key} commands`);
```

Comment:
I actually think the place you want to hook into this behavior is at the [Server](https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/core/sdam/server.js#L405) level. We will likely want a much better approach in the future, but until we have a coherent "command construction" pipeline, this will have to do. You can hook into `update` and `delete` (`remove`) there.

It's an implementation detail that all write operations go through the "bulk" code right now, and I'd prefer not to jam more into it